alf/algorithms/actor_critic_loss.py

```diff
@@ -111,13 +111,18 @@ def __call__(self, training_info: TrainingInfo, value):
         returns, advantages = self._calc_returns_and_advantages(
             training_info, value)
 
-        if self._debug_summaries and common.should_record_summaries():
+        def _summary():
             with tf.name_scope('ActorCriticLoss'):
                 tf.summary.scalar("values", tf.reduce_mean(value))
                 tf.summary.scalar("returns", tf.reduce_mean(returns))
                 tf.summary.scalar("advantages", tf.reduce_mean(advantages))
                 tf.summary.scalar("explained_variance_of_return_by_value",
                                   common.explained_variance(value, returns))
+                return tf.constant(True)
+
+        if self._debug_summaries:
+            tf.cond(common.should_record_summaries(),
+                    _summary, lambda: tf.constant(False))
```

Comment:
Is there any special meaning of returning tf.constant(True) or tf.constant(False)? Can we just return tf.zeros(()) for both cases?

Reply:
return value here has no meaning,  any shape of tensor is ok .  i just follow the source code of  [tf.summary.write](https://github.com/tensorflow/tensorflow/blob/8e423e3d56390671f0d954c90f4fd163ab02a9c1/tensorflow/python/ops/summary_ops_v2.py#L684-L701)